src/edit/selection.js

```diff
@@ -83,7 +83,7 @@ class SelectionState {
   readFromDOM() {
     if (!hasFocus(this.pm) || !this.domChanged()) return false
 
-    let {range, adjusted} = selectionFromDOM(this.pm.doc, this.range.head)
+    let {range, adjusted} = selectionFromDOM(this.pm.doc, this.range.head, this.pm.root)
```

Comment:
I would probably just change `selectionFromDOM` to take `this.pm` as its first argument, rather than `this.pm.doc`.